### PR TITLE
Modified some of the reporting keys and overhauled the wandb reporter

### PR DESCRIPTION
**Pull request by scarere**
# PR Type
Feature and Fix

# Short Description

I noticed an issue where data at the round level was not reported if the wandb reporter step type was step. This was initially the intended behaviour, but we were losing information which wasn't logged at the step level. Restructured the wandb reporter to fix this and also fixed a few things. Additionally I needed to change the metric key for fit_metrics and fit_losses since they were not the same at both the round, epoch and step level. Added some comments to clarify.

A future PR might look into having a single config file that defines all our reporting/logging metrics.

# Tests Added

Modified some of the tests to reflect the new reporting keys

### CODE REVIEW

**Review by emersodb — COMMENTED** [review]
Generally looks good. Just a few fairly minor comments.

File: fl4health/clients/basic_client.py [hunk 1]
```diff
@@ -307,15 +308,24 @@ def fit(self, parameters: NDArrays, config: Config) -> Tuple[NDArrays, int, dict
...
+        #   - Train by steps: round metrics/losses are computed using all samples from the round
+        #   - Train by epochs: round metrics/losses computed using only the samples from the final epoch of the round
+        #   - fit_round_metrics: Computed at the end of the round on the samples directly
+        #   - fit_round_losses: The average of the losses computed for each step.
+        #       * (Hence likely higher than the final loss of the round.)
         self.reports_manager.report(
             {
-                "fit_metrics": metrics,
-                "fit_losses": loss_dict,
+                "fit_round_metrics": metrics,  # Metrics computed
```
[the remark this answers is not in the document]
Comment (scarere) [reply fl4health/clients/basic_client.py:319]: Nope just a comment I forgot to delete. I was going to explain it inline but instead but a comment block above

File: fl4health/clients/basic_client.py [hunk 2]
```diff
@@ -349,7 +359,7 @@ def evaluate(self, parameters: NDArrays, config: Config) -> Tuple[float, int, Di
             self.setup_client(config)
 
         start_time = datetime.datetime.now()
-        current_server_round = narrow_dict_type(config, "current_server_round", int)
+        local_epochs, local_steps, current_server_round, _, _ = self.process_config(config)
```
[the remark this answers is not in the document]
Comment (scarere) [reply fl4health/clients/basic_client.py:362]: they're not, will replace with `_`

File: fl4health/clients/basic_client.py [hunk 3]
```diff
@@ -364,14 +374,19 @@ def evaluate(self, parameters: NDArrays, config: Config) -> Tuple[float, int, Di
...
+                "eval_round_start": str(start_time),
+                "eval_round_time_elapsed": str(elapsed),
+                "eval_round_end": str(end_time),
+                "fit_step": self.total_steps,
+                "fit_epoch": self.total_epochs,
+                "round": current_server_round,
             },
             current_server_round,
         )
+        # Have to report 3 times to make sure they get reported regardless of the reporting level
```
[the remark this answers is not in the document]
Comment (scarere) [reply fl4health/clients/basic_client.py:388]: This is also a comment i forgot to delete

**Comment by emersodb** [comment]
Good to go here I think

**Review by emersodb — APPROVED** [review]

**Pull request merged**